examples/rosettacode/24_game.nit

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env nit
+#
+# This file is part of NIT ( http://www.nitlanguage.org ).
+# This program is public domain
+
+# Task: 24 game
+# SEE: <http://rosettacode.org/wiki/24_game>
+
+# Export `operands` and `numbers` from string `operation`
+fun exportation(operation: String, numbers: Array[Int], operands: Array[Char]) do
+	var previous_char: nullable Char = null
+	var number: nullable Int = null
+	var negative = false
+
+	for i in operation.length.times do
+		var current_char = operation[i]
+		var current_int = current_char.to_i
+
+		if (previous_char == null or "-+/*".has(previous_char)) and current_char == '-' then
+			negative = true
+			continue
+		end
+
+		if current_int >= 0 and current_int <= 9 then
+			if number == null then
+				number = current_int
+			else
+				number = number * 10 + current_int
+			end
+		end
+
+		if negative and ("-+/*".has(current_char) or i == operation.length - 1) then
+			number = number - number * 2
+			negative = false
+		end
+
+		if ("-+/*".has(current_char) or i == operation.length - 1) and number != null then
+			numbers.add(number)
+			number = null
+		end
+
+		if not negative and "-+/*".has(current_char) then
+			operands.add(current_char)
+		end
+		previous_char = current_char
+	end
+end
+
+# Create `random_numbers` from 0 to 9
+fun random(random_numbers: Array[Int]) do
+	for i in [0..4[ do
+		var number = 0
+		while number == 0 do number = 10.rand
+		random_numbers.add(number)
+	end
+end
```

Comment:
This can be simplified to:
~~~
for i in [0..4[ do
    var number = 1 + 9.rand
    random_numbers.add number
end
~~~

Actually, the whole method could even be:
~~~
return [for i in 4.times do 1 + 9.rand]
~~~